doc/upgrade.md

```diff
@@ -0,0 +1,136 @@
+```
+
+and a crypttab entry like:
+
+```
+cr_home     /dev/sda2
+```
+
+the device "/dev/sda2" is searched in the devicegraph and the crypttab name "cr_home" is stored in its associated encryption device. So, in case that there is an encryption device over "/dev/sda2", for example "/dev/mapper/cr-auto-1", that encryption device will save "cr_auto" as alternative name.
+
+Once this information is saved in the encryption devices, when an encryption name is searched in the devicegraph, we can check if the searched named is the alternative name of any encryption device. So, for this example, when we search in the devicegraph the device by "/dev/mapper/cr_home", we can return "/dev/mapper/cr-auto-1".
+
+This problem was fixed for SLE-15, see [related PBI](https://trello.com/c/OGssTRzO/393-osdistribution-p2p1-1094963-opensuse-150-installation-or-upgrade-fails-if-disk-contains-encrypted-partition-luks).
+
+
+## Problem 2: device not found in inst-sys
+
+The second problem comes when the mounting action is peformed by using the encryption name read from the fstab file. As mentioned above, the fstab entry for an encryption device could be one of the following:
+
+```
+UUID=111-222-333                            /home   ext4    defaults 0 0  # mount_by = uuid
+/dev/mapper/cr_ata-111-part2                /home   ext4    defaults 0 0  # mount_by = device name
+LABEL=home_enc                              /home   ext4    defaults 0 0  # mount_by = volume label
+/dev/disk/by-id/dm-name-cr_ata-111-part2    /home   ext4    defaults 0 0
+/dev/disk/by-uuid/111-222-333               /home   ext4    defaults 0 0
+/dev/disk/by-label/home_enc                 /home   ext4    defaults 0 0
+```
+
+The fstab spec could be problematic when it includes the encryption device name as part of the device spec, that is:
+
+```
+/dev/mapper/cr_ata-111-part2                /home   ext4    defaults 0 0
+/dev/disk/by-id/dm-name-cr_ata-111-part2    /home   ext4    defaults 0 0
+```
+
+In such cases, the mount command would fail if the encryption device was probed with another name. For example, if the encryption device was probed with "cr-auto-1" name, the inst-sys will contain the following device names:
+
+```
+/dev/mapper/cr-auto-1
+/dev/disk/by-id/dm-name-cr-auto-1
+```
+
+So, when the fstab device tryied to be mounted, it will fail:
+
+```
+mount /dev/mapper/cr_ata-111-part2 /home  => fail device /dev/mapper/cr_ata-111-part2 does not exist
+```
+
+The solution here is to avoid to use the fstab spec when that device name is not safe enough. So, instead of using the fstab spec, the udev uuid name of the encryption device will be used due to that name exists in the inst-sys:
+
+```
+mount /dev/disk/by-uuid/111-222-333 /home
+```
+
+Note that this problem could not appear when upgrading a system generated with old storage. The old storage created the fstab entries with the option "nofail" for some mount points, for example "/" or "/home":
```

Comment:
I would say "this problem could not appear". I would say the symptom is different. Silently failing to mount the filesystem and performing the upgrade without it anyways is also a problem... or another side of the same problem.